contracts/src/CrossChainApplications/NativeTokenBridge/ERC20TokenSource.sol

```diff
@@ -0,0 +1,174 @@
+// (c) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// SPDX-License-Identifier: Ecosystem
+
+pragma solidity 0.8.18;
+
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@subnet-evm-contracts/interfaces/IWarpMessenger.sol";
+import "./IERC20TokenSource.sol";
+import "../../Teleporter/ITeleporterMessenger.sol";
+import "../../Teleporter/ITeleporterReceiver.sol";
+import "../../Teleporter/SafeERC20TransferFrom.sol";
+
+contract ERC20TokenSource is
+    ITeleporterReceiver,
+    IERC20TokenSource,
+    ReentrancyGuard
+{
+    using SafeERC20 for IERC20;
+
+    uint256 public constant MINT_NATIVE_TOKENS_REQUIRED_GAS = 150_000; // TODO this is a placeholder
+    bytes32 public immutable currentBlockchainID;
+    bytes32 public immutable destinationBlockchainID;
+    address public immutable nativeTokenDestinationAddress;
+    address public immutable erc20ContractAddress;
+
+    // Used for sending an receiving Teleporter messages.
+    ITeleporterMessenger public immutable teleporterMessenger;
+
+    constructor(
+        address teleporterMessengerAddress,
+        bytes32 destinationBlockchainID_,
+        address nativeTokenDestinationAddress_,
+        address erc20ContractAddress_
```

Comment:
I'd prefer to name these parameters using abbreviated forms (e.g. `dstBlockchainID`), rather than the trailing underscore to make them distinct from the contract state variables they set. To me, the trailing underscore implies some special treatment of the variable.

Reply:
This is what's recommended in the [solidity style guide](https://docs.soliditylang.org/en/v0.8.19/style-guide.html#avoiding-naming-collisions). Happy to change this if the team doesn't like it though

Reply:
Nope, sounds good to me! I didn't realize this was Solidity convention.